ls1intum/Artemis
# Plagiarism detection configuration

### Checklist

### Motivation and Context
Currently, the options for plagiarism detection are fixed and the same for every run. This PR adds configuration options to the plagiarism page to allow instructors to change these values before starting the detection.

### Description
I slightly changed the layout of the plagiarism page header and made the options panel collapsable. By clicking on the page title, the user can configure the options to his needs and hide them afterwards to maximize the screen area available for inspection.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Go to the Exercises of a course
4. Select a programming, text, or modeling exercise
5. Click 'Check Plagiarism'

On the plagiarism page, make sure that the following works:

* Clicking on the page title should show/hide the plagiarism configuration options
* Available options for **text and modeling exercises**: similarity threshold, minimum score, minimum size
* Available options for **programming exercises**: generate JPlag report, similarity threshold, minimum score
* Tooltips should be available for similarity threshold, minimum score, and minimum size
* Minimum Score and Minimum Size, if available, are optional and must be enabled by clicking on the checkbox to take effect
* Make sure that for each exercise type the options have the desired effect on the result

### Test Coverage
TODO

### Screenshots
<img width="1270" alt="Screenshot 2021-01-13 at 11 10 47" src="https://user-images.githubusercontent.com/21973908/104438697-93e76880-5590-11eb-9d66-817abe38d980.png">
